dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/data/file/RumFileWriter.kt

```diff
@@ -32,18 +39,44 @@ internal class RumFileWriter(
 
     override fun writeData(data: ByteArray, model: RumEvent): Boolean {
         val writeDataSuccess = super.writeData(data, model)
-        if (writeDataSuccess && model.event is ViewEvent) {
-            if (!lastViewEventFile.exists()) {
-                lastViewEventFile.createNewFile()
+        if (writeDataSuccess) {
+            val event = model.event
+            when (event) {
+                is ViewEvent -> persistViewEvent(data)
+                is ActionEvent -> notifyEventSent(event.view.id, EventType.ACTION)
+                is ResourceEvent -> notifyEventSent(event.view.id, EventType.RESOURCE)
+                is ErrorEvent -> notifyEventSent(event.view.id, EventType.ERROR)
+                is LongTaskEvent -> notifyEventSent(event.view.id, EventType.LONG_TASK)
             }
-            // persist the serialised ViewEvent in the NDK crash data folder
-            fileHandler.writeData(lastViewEventFile, data)
+        } else {
+            System.err.println("writeDataFailed!")
         }
         return writeDataSuccess
     }
 
     // endregion
 
+    // region Internal
+
+    private fun persistViewEvent(data: ByteArray) {
+        if (!lastViewEventFile.exists()) {
+            lastViewEventFile.createNewFile()
+        }
+        // persist the serialised ViewEvent in the NDK crash data folder
+        fileHandler.writeData(lastViewEventFile, data)
+    }
+
+    private fun notifyEventSent(viewId: String, eventType: EventType) {
```

Comment:
should we use `sent` in the naming here? because at this point it is not yet sent, but only persisted

Reply:
makes sense. anyway, after further exploration of code base I see that term `sent` is used in many places, so it makes sense for the consistency as well.